tasks/main.yml

```diff
@@ -1 +1,30 @@
 # role tasks
+- name: Rebooting in {{ reboot_seconds_before_reboot }} seconds
+  command: >
+    systemd-run --on-active="{{ reboot_seconds_before_reboot }}"
+    systemctl reboot
+  become: "{{ reboot_become }}"
+  become_user: "{{ reboot_become_user }}"
+  tags: skip_ansible_lint
+
+- name: >
+    Reboot command has been issued, waiting a maximum of
+    {{ reboot_wait_for_connection_timeout }} seconds to reconnect.
+  wait_for_connection:
+    delay: "{{ reboot_wait_for_connection_delay }}"
+    timeout: "{{ reboot_wait_for_connection_timeout }}"
+
+- name: Get the system uptime before reboot
+  set_fact:
+    reboot_before: "{{ ansible_uptime_seconds }}"
+
+- name: Get the system uptime after reboot
+  setup:
+    gather_subset: '!all,!min,hardware'
+
+- name: Provide feedback on failure
+  fail:
```

Comment:
@seandst 's outdated comments about using `assert` instead of `fail` are still relevant here.